tests/test_brownian_interval.py

```diff
@@ -0,0 +1,311 @@
+        sample1 = bm(ta, return_U=return_U, return_A=return_A)
+        sample2 = bm(tb, return_U=return_U, return_A=return_A)
+        sample3 = bm(ta, tb, return_U=return_U, return_A=return_A)
+        shapes = []
+        A_shapes = []
+        for sample in (sample1, sample2, sample3):
+            if return_U:
+                if return_A:
+                    W1, U1, A1 = sample
+                    shapes.append(W1.shape)
+                    shapes.append(U1.shape)
+                    A_shapes.append(A1.shape)
+                else:
+                    W1, U1 = sample
+                    shapes.append(W1.shape)
+                    shapes.append(U1.shape)
+            else:
+                if return_A:
+                    W1, A1 = sample
+                    shapes.append(W1.shape)
+                    A_shapes.append(A1.shape)
+                else:
+                    W1 = sample
+                    shapes.append(W1.shape)
+
+        for shape_ in shapes:
+            assert shape_ == shape
+        for shape_ in A_shapes:
+            assert shape_ == A_shape
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation, return_U, return_A", _levy_returns())
+def test_determinism_simple(device, levy_area_approximation, return_U, return_A):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    ta, tb, bm = _setup(device, levy_area_approximation, (SMALL_BATCH_SIZE, D))
+    vals = [bm(ta, tb, return_U=return_U, return_A=return_A) for _ in range(REPS)]
+    for val in vals[1:]:
+        if torch.is_tensor(val):
+            val = (val,)
+        if torch.is_tensor(vals[0]):
+            val0 = (vals[0],)
+        else:
+            val0 = vals[0]
+        for v, v0 in zip(val, val0):
+            assert (v == v0).all()
+
+
+@pytest.mark.parametrize("device", devices)
+@pytest.mark.parametrize("levy_area_approximation, return_U, return_A", _levy_returns())
+def test_determinism_large(device, levy_area_approximation, return_U, return_A):
+    if device == gpu and not torch.cuda.is_available():
+        pytest.skip(msg="CUDA not available.")
+
+    ta, tb, bm = _setup(device, levy_area_approximation, (SMALL_BATCH_SIZE, D))
+    cache = {}
+    for _ in range(LARGE_REPS):
+        ta_ = torch.rand_like(ta)
+        tb_ = torch.rand_like(tb)
+        ta_, tb_ = min(ta_, tb_), max(ta_, tb_)
+        val = bm(ta_, tb_, return_U=return_U, return_A=return_A)
+        if torch.is_tensor(val):
+            val = (val,)
+        cache[ta_, tb_] = tuple(v.detach().clone() for v in val)
+
+    cache2 = {}
+    for ta_, tb_ in cache:
+        val = bm(ta_, tb_, return_U=return_U, return_A=return_A)
+        if torch.is_tensor(val):
+            val = (val,)
+        cache2[ta_, tb_] = tuple(v.detach().clone() for v in val)
+
+    for ta_, tb_ in cache:
+        for v1, v2 in zip(cache[ta_, tb_], cache2[ta_, tb_]):
+            assert (v1 == v2).all()
```

Comment:
Could we do `torch.allclose` since these are floating point tensors?

Reply:
That would only test closeness; here I specifically want to test for equality, i.e. perfect determinism.

Reply:
SGTM.